07_yaml/test/test_exercicio.rb

```diff
@@ -15,4 +17,43 @@ class TestExercicio < Minitest::Test
   # 1. Quando o nome for Joao, a matrícula deverá ser atualizada para '0004'
   # 2. Quando a matricula for '0001', o nome deverá ser atualizado para 'Ze Felipe'
   # 3. Quando a matricula for '0003', deve ser removido da lista
+
+
+  def test_list
+    list = [
+    { nome: 'Ze', matricula: '0001' },
+    { nome: 'Joao', matricula: '0002' },
+    { nome: 'Maria', matricula: '0003' }
+    ]
+
+    File.open('07_yaml/test/students.yml', 'w') do |file|
+      file.write(list.to_yaml)
+    end
+
+    list_read = File.read('07_yaml/test/students.yml')
+    list_data = YAML.load(list_read)
+
+    assert_equal list, list_data
+
+    list_data.each do |student|
+      if student[:nome] == 'Joao'
+        student[:matricula] = '0004'
+      end
+
+      if student[:matricula] == '0001'
+      student[:nome] = 'Ze Felipe'
+      elsif student[:matricula] == '0003'
+      list_data.delete(student)
+      end
```

Comment:
[Not blocked]

Só identação 😄 
```suggestion
      if student[:matricula] == '0001'
        student[:nome] = 'Ze Felipe'
      elsif student[:matricula] == '0003'
        list_data.delete(student)
      end
```